scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/InMemoryConnectionStore.java

```diff
@@ -272,4 +294,31 @@ public void handshakeCompleted(final InetSocketAddress peer) {
 	public void handshakeFailed(final InetSocketAddress peer, Throwable error) {
 		// nothing to do
 	}
+
+	/**
+	 * Check if the connection is expired, if this is the case remove it from
+	 * the store.
+	 * 
+	 * @return return the connection only if the connection has no session
+	 *         established or if the established session is not expired.
+	 */
+	private Connection checkValidity(Connection connection) {
+		if (connection == null) {
+			return null;
+		}
+		if (!connection.hasEstablishedSession()) {
+			if (!connection.hasSessionTicket()) {
+				return connection;
+			} else if (!connection.getSessionTicket().isExpired(sessionLifeTime)) {
+				return connection;
```

Comment:
I think, this check is better done in `deepfind` and return null, when expired.

Connections with tickets live only very short in `DTLSConnector.resumeExistingSession`.
They are not stored in the connection store and therefore need not to be removed from that.